query/src/main/scala/filodb/query/PromCirceSupport.scala

```diff
@@ -40,22 +44,34 @@ object PromCirceSupport {
 
   implicit val decodeFoo: Decoder[DataSampl] = new Decoder[DataSampl] {
     final def apply(c: HCursor): Decoder.Result[DataSampl] = {
-      val tsResult = c.downArray.as[Long]
-      val rightCurs = c.downArray.right
-      if (rightCurs.focus.get.isObject) {
-        for { timestamp <- tsResult
-              buckets   <- rightCurs.as[Map[String, Double]] } yield {
-          HistSampl(timestamp, buckets)
-        }
-      } else {
-        for { timestamp <- tsResult
-              value     <- rightCurs.as[String] } yield {
-          Sampl(timestamp, value.toDouble)
-        }
+      c.downField("cardinality").focus match {
+        case Some(_) =>
+          // TODO: Not the best way to find if this is a label cardinality response, is there a better way?
```

Comment:
Maybe there's no getting around a "response-type" field?

Reply:
Can't find something that checks for existence of a field without actually reading it. That would be ideal